includes/Generator/OrderAttribution.php

```diff
@@ -0,0 +1,300 @@
+            case 'referral':
+                $refferal = array(
+                    'https://woo.com/',
+                    'https://facebook.com',
+                    'https://twitter.com',
+                );
+                return $refferal[ array_rand( $refferal ) ];
+            case 'typein':
+                return '';
+            case 'admin':
+                return '';
+            default:
+                return '';
+        }
+    }
+
+    /**
+     * Get a random utm medium.
+     *
+     * @return string The utm medium.
+     */
+    public static function get_random_utm_medium() {
+        $utm_mediums = array(
+            'referral',
+            'cpc',
+            'email',
+            'social',
+            'organic',
+            'unknown'
+        );
+
+        return $utm_mediums[ array_rand( $utm_mediums ) ];
+    }
+
+    /**
+     * Get the origin.
+     *
+     * @param string $source_type The source type.
+     * @param string $source The source.
+     *
+     * @return string The origin.
+     */
+	public static function get_origin( string $source_type, string $source ) {
+		// Set up the label based on the source type.
+		switch ( $source_type ) {
+			case 'utm':
+				return 'Source: ' . $source;
+			case 'organic':
+				return 'Organic: ' . $source;
+			case 'referral':
+				return 'Referral: ' . $source;
+			case 'typein':
+				return 'Direct';
+			case 'admin':
+				return 'Web admin';
+			default:
+				return 'Unknown';
+		}
+	}
+
+    /**
+     * Get random source type.
+     *
+     * @return string The source type.
+     */
+    public static function get_source_type() {
+        $source_types = array(
+            'typein',
+            'organic',
+            'referral',
+            'utm',
+            'admin',
+            'unknown'
+        );
+
+        return $source_types[ array_rand( $source_types ) ];
+    }
+
+    /**
+     * Get random source based on the source type.
+     *
+     * @param string $source_type The source type.
+     * @return string The source.
+     */
+    public static function get_source( $source_type ) {
+        switch ( $source_type ) {
+            case 'typein':
+                return '';
+            case 'organic':
+                $organic = array(
+                    'google',
+                    'bing',
+                    'yahoo',
+                );
+                return $organic[ array_rand( $organic ) ];
+            case 'referral':
+                $refferal = array(
+                    'woo.com',
+                    'facebook.com',
+                    'twitter.com',
+                );
+                return $refferal[ array_rand( $refferal ) ];
+            case 'social':
+                $social = array(
+                    'facebook.com',
+                    'twitter.com',
+                    'instagram.com',
+                    'pinterest.com',
+                );
+                return $social( array_rand( $social ) );
```

Comment:
🔎  
```suggestion
                return $social[ array_rand( $social ) ];
```

Although I'm not sure `social` as a source exists anyway ... all those pages are tagged simply as `referral` by Sourcebuster.